src/silicone/utils.py

```diff
@@ -526,3 +526,17 @@ def _construct_consistent_values(aggregate_name, components, db_to_generate):
     use["unit"] = units.iloc[0]
     use["variable"] = aggregate_name
     return pyam.IamDataFrame(use)
+
+
+def _remove_deprecation_warnings(caplog_warnings):
```

Comment:
We should be able to get rid of this and use https://docs.pytest.org/en/stable/warnings.html#pytest-mark-filterwarnings instead e.g. `@pytest.mark.filterwarnings("ignore:.*deprecated.*:DeprecationWarning")`

Reply:
So it's unfortunately not a type of deprecation warning, simply a pyam warning with a string that begins a particular way. I had thought that something like this could work with @pytest.mark.filterwarnings("ignore:This method is deprecated*") but I haven't managed to get the syntax right for that.

Reply:
I'll raise an issue in pyam about this now.

For now, try `@pytest.mark.filterwarnings("ignore:.*This method is deprecated.*")` (You need the `.*` to capture using regex, rather than glob where it would just be `*`, checkout regex101.com)

Reply:
I'm not really having any luck with that either.

Reply:
Right so turns out they're not actually warnings (because pyam's warnings and logging is broken, see https://github.com/IAMconsortium/pyam/issues/448 and https://github.com/IAMconsortium/pyam/issues/449). We have to use caplog's ability to change logging levels instead. Fix in #120.